rskj-core/src/test/java/co/rsk/rpc/modules/debug/trace/call/CallTracerTest.java

```diff
@@ -69,7 +66,43 @@ void retrieveSimpleStorageContractCreationTrace() throws Exception {
         assertEquals("CREATE", traceResult.getType());
         assertEquals("0x608060405234801561000f575f80fd5b506101438061001d5f395ff3fe608060405234801561000f575f80fd5b5060043610610034575f3560e01c80636057361d146100385780636d4ce63c14610054575b5f80fd5b610052600480360381019061004d91906100ba565b610072565b005b61005c61007b565b60405161006991906100f4565b60405180910390f35b805f8190555050565b5f8054905090565b5f80fd5b5f819050919050565b61009981610087565b81146100a3575f80fd5b50565b5f813590506100b481610090565b92915050565b5f602082840312156100cf576100ce610083565b5b5f6100dc848285016100a6565b91505092915050565b6100ee81610087565b82525050565b5f6020820190506101075f8301846100e5565b9291505056fea2646970667358221220271ba6597ab51821beed677d25c76e319892db25c1f66b3dc76e547fdc1fd0e164736f6c63430008140033", traceResult.getInput());
         assertEquals(account.getAddress().toJsonString(), traceResult.getFrom());
+    }
+
+    @Test
+    void internalOoGErrorTrace() throws Exception {
+        DslParser parser = DslParser.fromResource("dsl/call_tracer_internal_oog.txt");
+        ReceiptStore receiptStore = new ReceiptStoreImpl(new HashMapDB());
+        World world = new World(receiptStore);
+        ExecutionBlockRetriever executionBlockRetriever = Mockito.mock(ExecutionBlockRetriever.class);
+        Web3InformationRetriever web3InformationRetriever = new Web3InformationRetriever(world.getTransactionPool(), world.getBlockChain(), world.getRepositoryLocator(), executionBlockRetriever);
+
+        WorldDslProcessor processor = new WorldDslProcessor(world);
+        processor.processCommands(parser);
+
+        TransactionReceipt contractTransactionReceipt = world.getTransactionReceiptByName("tx02");
 
+        CallTracer callTracer = new CallTracer(world.getBlockStore(), world.getBlockExecutor(), web3InformationRetriever, receiptStore, world.getBlockChain());
+
+        JsonNode result = callTracer.traceTransaction(contractTransactionReceipt.getTransaction().getHash().toJsonString(), new TraceOptions());
+
+        assertNotNull(result);
+
+        TxTraceResult traceResult = objectMapper.treeToValue(result, TxTraceResult.class);
+
+        assertNotNull(traceResult);
+
+        assertError(traceResult);
+
+        assertFalse(traceResult.getCalls().isEmpty());
+        assertError(traceResult.getCalls().get(0));
+
+        assertFalse(traceResult.getCalls().get(0).getCalls().isEmpty());
+        assertError(traceResult.getCalls().get(0).getCalls().get(0));
     }
 
-}
\ No newline at end of file
+    private static void assertError(TxTraceResult trace) {
```

Comment:
I would suggest a new name for this function, something like "assertOOGError" or something like that, as it is an specific case. WDYT?